src/core/public/application/types.ts

```diff
@@ -232,18 +235,46 @@ export interface App<HistoryLocationState = unknown> {
    * ```
    */
   exactRoute?: boolean;
+
+  /**
+   * Array of links that represent secondary in-app locations for the given app.
+   *
+   * Can be updated using the {@link App.updater$} observable. See {@link AppSubLink} for more details.
+   */
+  subLinks?: AppSubLink[];
 }
 
+/**
+ * Input type for registering secondary in-app locations for an application
+ *
+ * Used to populate navigational search results (where available).
+ * @public
+ */
+export type AppSubLink = { id: string; title: string } & (
+  | { path: string; subLinks?: AppSubLink[] }
+  | { path?: string; subLinks: AppSubLink[] }
```

Comment:
Will there be cases where a sublink can have both an 'existence' ( a `path`) and 'children' (`subLinks`)?

Reply:
It could in theory I think, yes. For instance, there may be an overview page for that section of the hierarchy, but also have sublinks inside of it. I typed it this way to make it clear that it has to have at least one of `path` or `subLinks`.